### ISSUE TITLE
Deprecation version on Python 3.11: ssl.SSLContext() without protocol argument is deprecated

### ISSUE THREAD

**Issue by ad-m-ss** [aws/amazon-redshift-python-driver#196]
## Driver version

`master`

## Redshift version

N/A

## Client Operating System

N/A

## Python version

3.11

## Table schema

N/A

## Problem description

1. Expected behaviour:
2. Actual behaviour:
3. Error message/stack trace:

```
/usr/local/lib/python3.11/site-packages/redshift_connector/__init__.py:340: in connect
    return Connection(
/usr/local/lib/python3.11/site-packages/redshift_connector/core.py:603: in __init__
    ssl_context: SSLContext = SSLContext()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'ssl.SSLContext'>, protocol = None, args = (), kwargs = {}

    def __new__(cls, protocol=None, *args, **kwargs):
        if protocol is None:
>           warnings.warn(
                "ssl.SSLContext() without protocol argument is deprecated.",
                category=DeprecationWarning,
                stacklevel=2
            )
E           DeprecationWarning: ssl.SSLContext() without protocol argument is deprecated.
```
4. Any other details that can be helpful:

https://github.com/aws/amazon-redshift-python-driver/blob/62a4d4e364055e3d3d6c0d334258ca556298a9b8/redshift_connector/core.py#L654

**Comment by Brooke-white** [comment]
Hi @ad-m-ss, thank you for identifying this deprecation warning. We plan to include resolution for it in our December release of redshift-connector.

**Comment by ad-m-ss** [comment]
Sounds good. 

Additional information:
* We reproduced that on Python 3.10, too.
* There is one more deprecation: `ssl.PROTOCOL_TLS is deprecated`

Suggestion:
* Could we update test / CI to block any warnings? Disallowing warnings allows you to detect such things and makes it so that any new warning must be explicitly reviewed (at least before being added to exceptions with accepted warnings), rather than being ignored in the console.

**Comment by Brooke-white** [comment]
Hi @ad-m-ss , a fix can be found in 2.0.918. The team will work to address blocking on warnings in CI.

**Comment by robmcd** [comment]
Still seeing this warning in redshift-connector 2.1.5:

```
C:\.venv\Lib\site-packages\redshift_connector\core.py:654: DeprecationWarning:

ssl.SSLContext() without protocol argument is deprecated.

C:\.venv\Lib\site-packages\redshift_connector\core.py:654: DeprecationWarning:

ssl.PROTOCOL_TLS is deprecated
```
